src/main/java/noppes/npcs/scripted/entity/ScriptPlayer.java

```diff
@@ -192,6 +204,14 @@ public void addFactionPoints(int faction, int points){
 		PlayerData data = PlayerDataController.instance.getPlayerData(player);
 		data.factionData.increasePoints(faction, points, player);
 	}
+	/**
+	 * @param faction The faction id
+	 * @param points The new point value for this faction
+	 */
+	public void setFactionPoints(int faction, int points) {
+		PlayerData data = PlayerDataController.instance.getPlayerData(player);
+		data.factionData.increasePoints(faction, points-getFactionPoints(faction), player);
+	}
```

Comment:
Same for this, all new functions to do with Factions should be retrieved by in a player data handler.

Reply:
would the addFactionPoints and getFactionPoints functions be moved to this data handler as well? And the Quest functions?

Reply:
They'd be in there too. For the sake of people's scripts not breaking, the functions will be in two places, but the javadoc documentation I'm making in the interfaces won't mention it being in IPlayer. Those functions will be marked as @Deprecated (unsupported), but people will still be able to use them with their old scripts.